backend/internal/handlers/user.go

```diff
@@ -152,6 +152,59 @@ func VerifyToken(ctx *gin.Context) {
 	ctx.JSON(http.StatusAccepted, gin.H{"message": "Token matches"})
 }
 
+// Get the current authenticated user
+func GetCurrentUser(ctx *gin.Context) {
+	token, exists := ctx.Get("sessionToken")
+	fmt.Println("This is the token from the context")
+	fmt.Println(token)
+
+	if !exists {
+		fmt.Println("error on line 161")
+		ctx.JSON(
+			http.StatusUnauthorized,
+			gin.H{
+				"error": "Unauthorized",
+			},
+		)
+		return
+	}
+
+	tokenStr, ok := token.(*string)
+	if !ok || tokenStr == nil {
+		fmt.Println("error on line 173")
+		fmt.Println("This is the token string")
+		fmt.Println(tokenStr)
+		fmt.Println("This is the ok object")
+		fmt.Println(ok)
+		ctx.JSON(
+			http.StatusUnauthorized,
+			gin.H{
+				"error": "Invalid token",
+			},
+		)
+		return
+	}
```

Comment:
Remember not using Get, but instead using MustGet, and also applying .(*string) on the same line so that you don't do double error checks cause that's long.